components/content/Icon/Icon.stories.ts

```diff
@@ -0,0 +1,254 @@
+`;
+
+export default {
+  component: 'diamond-icon',
+  parameters: {
+    docs: {
+      description: {
+        component: description,
+      },
+    },
+  },
+  argTypes: {
+    shape: {
+      control: {
+        type: 'radio',
+      },
+      options: ['default', 'circle'],
+    },
+  },
+};
+
+export const Icon: StoryObj = {
+  render: (args) => html`
+    <diamond-icon shape="${args.shape}">
+      <svg fill="none" viewBox="0 0 24 24" height="24" width="24">
+        <path
+          d="M2 6C2 4.89543 2.89543 4 4 4H20C21.1046 4 22 4.89543 22 6V18C22 19.1046 21.1046 20 20 20H4C2.89543 20 2 19.1046 2 18V6ZM5.51859 6L12 11.6712L18.4814 6H5.51859ZM20 7.32877L12.6585 13.7526C12.2815 14.0825 11.7185 14.0825 11.3415 13.7526L4 7.32877V18H20V7.32877Z"
+          fill="currentColor"
+        ></path>
+      </svg>
+    </diamond-icon>
+    This is some text next to the icon
+  `,
+};
+
+Icon.args = {
+  shape: 'default',
+};
+
+const sizes = ['h1', 'h2', 'h3', 'h4', 'md', 'default', 'sm', 'xs'];
+
+export const Sizes: StoryObj = {
+  render: () => html`
+    ${sizes.map(
+      (size) => html`
+        <p class="diamond-text-size-${size}">
+          <diamond-icon>
+            <svg fill="none" viewBox="0 0 24 24" height="24" width="24">
+              <path
+                d="M2 6C2 4.89543 2.89543 4 4 4H20C21.1046 4 22 4.89543 22 6V18C22 19.1046 21.1046 20 20 20H4C2.89543 20 2 19.1046 2 18V6ZM5.51859 6L12 11.6712L18.4814 6H5.51859ZM20 7.32877L12.6585 13.7526C12.2815 14.0825 11.7185 14.0825 11.3415 13.7526L4 7.32877V18H20V7.32877Z"
+                fill="currentColor"
+              ></path>
+            </svg>
+          </diamond-icon>
+          The quick brown fox jumps over the lazy dog.
+        </p>
+      `,
+    )}
+  `,
+};
+
+Sizes.parameters = {
+  docs: {
+    description: {
+      story:
+        'Icons inherit their size from the current text size or can be sized using text size classes.',
+    },
+  },
+};
+
+export const IconInAGrid: StoryObj = {
+  render: () => html`
+    <diamond-grid align-items="baseline" gap="xs">
+      <diamond-grid-item>
+        <diamond-icon class="diamond-text-size-h3">
+          <svg fill="none" viewBox="0 0 24 24" height="24" width="24">
+            <path
+              d="M2 6C2 4.89543 2.89543 4 4 4H20C21.1046 4 22 4.89543 22 6V18C22 19.1046 21.1046 20 20 20H4C2.89543 20 2 19.1046 2 18V6ZM5.51859 6L12 11.6712L18.4814 6H5.51859ZM20 7.32877L12.6585 13.7526C12.2815 14.0825 11.7185 14.0825 11.3415 13.7526L4 7.32877V18H20V7.32877Z"
+              fill="currentColor"
+            ></path>
+          </svg>
+        </diamond-icon>
+      </diamond-grid-item>
+      <diamond-grid-item grow shrink>
+        <h3 class="diamond-spacing-bottom-none">Title text</h3>
+      </diamond-grid-item>
+    </diamond-grid>
+  `,
+};
+
+IconInAGrid.parameters = {
+  docs: {
+    description: {
+      story:
+        'Icons in a grid can be aligned to the text with the baseline alignment.',
```

Comment:
Does this work when the text wraps onto multiple lines?

```
Icon | Text
        | Text
```

I have a [icon text component in RL](https://github.com/etchteam/recycling-locator/tree/main/src/components/composition/IconText) which sees regular use

Reply:
<img width="626" alt="Screenshot 2024-05-03 at 04 13 36" src="https://github.com/etchteam/diamond-ui/assets/800251/4390d77f-905b-4662-aafe-127b3c150e14">

Yep, looks great